addons/stock_picking_batch/wizard/stock_picking_to_batch.py

```diff
@@ -29,3 +30,6 @@ def attach_pickings(self):
             batch = self.batch_id
 
         pickings.write({'batch_id': batch.id})
+        # you have to set some pickings to batch before confirm it.
+        if not self.is_create_draft and self.mode == 'new':
```

Comment:
Nitpick:
```suggestion
        if self.mode == 'new' and not self.is_create_draft:
```
This makes the line more logical to read and skips checking `is_create_draft` if the mode != new